### PR TITLE
Add new refactor lesson

### PR DESCRIPTION
**Pull request by tmikeschu**
@case-eee @neight-allen here is a new structure and approach to refactor tractor. Let me know if you want any changes!

### CODE REVIEW

File: module4/lessons/refactor-workshop.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+Each group present their "smellyscript" to practice technical communication (3 minutes).
+
+## Review presented topics
+
+## Code-along 1-2 topics
+
+## Paired Practice
+
+In your assigned pair:
```
Comment (case-eee) [inline module4/lessons/refactor-workshop.md]: are these different than their QS partners? just want to clarify for my sake!

File: module4/lessons/refactor-workshop.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+* Checkout a refactoring branch for Person A's project
+* Work on refactoring Person A's code snippet from their presentation for 20 minutes.
```
Comment (case-eee) [inline module4/lessons/refactor-workshop.md]: they're pairing together on one person's code snippet then they're going to swap and pair together for another 20 minutes on the other person's code snippet, correct?

File: module4/lessons/refactor-workshop.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+* Repeat for Person B
+
+## Review
+
+With your Quantified Self partner, discuss the following (3 minutes):
+
+  * How your workshop pair approached refactoring
+  * Your main takeaway from today's workshop
```
Comment (case-eee) [inline module4/lessons/refactor-workshop.md]: we can also ask a question on the survey for friday about what's one thing they learned during the JS refactoring class 👍

File: module4/lessons/refactor-workshop.md (hunk 1 above)
Comment (tmikeschu) [inline module4/lessons/refactor-workshop.md]: @case-eee yep! I have them made.

File: module4/lessons/refactor-workshop.md (hunk 2 above)
Comment (tmikeschu) [inline module4/lessons/refactor-workshop.md]: Yes: Person A's code for 20 minutes, and Person B's code for 20 minutes.

**Comment by tmikeschu** [comment]
Thanks for the feedback, @case-eee !

File: module4/lessons/refactor-workshop.md (hunk 3 above)
Comment (tmikeschu) [inline module4/lessons/refactor-workshop.md]: @case-eee will you add that or should I?

Comment (case-eee) [inline module4/lessons/refactor-workshop.md]: i've got it - as long as you send me the question, i'll go ahead and add it 😄

**Pull request merged**